Update helm fields (#176)
k3s-io/docs: docs/helm/helm.md

      enabled: true
```

> Another example is deploying [cert-manager](https://cert-manager.io/) with authentication:

Reviewer: This is a great example, but I think we should replace cert-manager with a more contrived example that is clearly pulling from a private chart repo, lest someone try to copy-paste this example to deploy cert-manager and open an issue complain that it's broken due to bad creds and CA cert.

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: cert-manager
---
apiVersion: helm.cattle.io/v1
kind: HelmChart

> metadata:
>   namespace: kube-system
>   name: cert-manager
> spec:
>   targetNamespace: cert-manager
>   createNamespace: true
>   version: v1.11.0
>   chart: cert-manager
>   repo: https://charts.jetstack.io
>   authSecret:
>     name: jetstack-auth
>   repoCAConfigMap:
>     name: jetstack-ca
>   set:
>     installCRDs: "true"

Reviewer: ```suggestion
metadata:
  namespace: kube-system
  name: example-app
spec:
  targetNamespace: example
  createNamespace: true
  version: v1.2.3
  chart: example-app
  repo: https://secure-repo.example.com
  authSecret:
    name: example-repo-auth
  repoCAConfigMap:
    name: example-repo-ca
  valuesContent: |-
    image:
      tag: v1.2.2
```

---
apiVersion: v1
kind: Secret
metadata:
  namespace: kube-system

>   name: jetstack-auth

Reviewer: ```suggestion
  name: example-repo-auth
```

type: kubernetes.io/basic-auth
stringData:
  username: user
  password: pass
---
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: kube-system

>   name: jetstack-ca

Reviewer: ```suggestion
  name: example-repo-ca
```

| metadata.name |   | Helm Chart name | NAME |
| spec.chart |   | Helm Chart name in repository, or complete HTTPS URL to chart archive (.tgz) | CHART |
| spec.targetNamespace | default | Helm Chart target namespace | `--namespace` |
| spec.createNamespace | false | Create target namespace if not present | `--create-namespace` |
| spec.version |   | Helm Chart version (when installing from repository) | `--version` |
| spec.repo |   | Helm Chart repository URL | `--repo` |

> | spec.repoCA | | Specify the certificates of HTTPS-enabled servers | `--ca-file` |
> | spec.repoCAConfigMap |  | Values of all keys in ConfigMap will be merged with repoCA | `--ca-file` |

Reviewer: ```suggestion
| spec.repoCA | | Verify certificates of HTTPS-enabled servers using this CA bundle. Should be a string containing one or more PEM-encoded CA Certificates. | `--ca-file` |
| spec.repoCAConfigMap | | Reference to a ConfigMap containing CA Certificates to be be trusted by Helm. Can be used along with or instead of `repoCA` | `--ca-file` |
```

| spec.helmVersion | v3 | Helm version to use (`v2` or `v3`) |  |
| spec.bootstrap | False | Set to True if this chart is needed to bootstrap the cluster (Cloud Controller Manager, etc) |  |
| spec.set |   | Override simple default Chart values. These take precedence over options set via valuesContent. | `--set` / `--set-string` |
| spec.jobImage |   | Specify the image to use when installing the helm chart. E.g. rancher/klipper-helm:v0.3.0 . | |
| spec.backOffLimit | 1000 | Specify the number of retries before considering a job failed. | |
| spec.timeout | 300s | Timeout for Helm operations, as a [duration string](https://pkg.go.dev/time#ParseDuration) (`300s`, `10m`, `1h`, etc) | `--timeout` |
| spec.failurePolicy | reinstall | Set to `abort` which case the Helm operation is aborted, pending manual intervention by the operator. | |

> | spec.authSecret | | Pass basic username/pass auth to Chart repos via `kubernetes.io/basic-auth` | |
> | spec.authPassCredentials | false | Pass credentials to all domains | `--pass-credentials` |
> | spec.dockerRegsistrySecret | | Point to mounted docker auth secret via `kubernetes.io/dockerconfigjson`. Used to authenticate OCI-compliant Docker registries | |

Reviewer: ```suggestion
| spec.authSecret | | Reference to Secret of type `kubernetes.io/basic-auth` holding Basic auth credentials for the Chart repo. | |
| spec.authPassCredentials | false | Pass Basic auth credentials to all domains. | `--pass-credentials` |
| spec.dockerRegistrySecret | | Reference to Secret of type `kubernetes.io/dockerconfigjson` holding Docker auth credentials for the OCI-based registry acting as the Chart repo. | |
```
